supervisor/supervisor.go

```diff
@@ -451,10 +451,14 @@ func (s *Supervisor) getExecSyncChannel(containerID, pid string) chan struct{} {
 	return ch
 }
 
-func (s *Supervisor) getExecSyncMap(containerID string) map[string]chan struct{} {
+func (s *Supervisor) getExecSyncOneChannel(containerID string) chan struct{} {
 	s.containerExecSyncLock.Lock()
 	defer s.containerExecSyncLock.Unlock()
-	return s.containerExecSync[containerID]
+
+	for _, ch := range s.containerExecSync[containerID] {
```

Comment:
maybe use an if here:
```go
if ch, ok := s.containerExecSync[containerID]; ok {
   return ch
}
```

Reply:
Use s.containerExecSync[containerID], we can get a map, not a chan.
if we use a map and list it, we need lock because of concurrent read and write, but we can't lock, when we wait a chan, it's block, and it make deleteExecSyncChannel not return. I have no good way to do get a chan if containerExecSync[containerID],  so I use range.
```
package main


import (
        "fmt"
)

func main(){
        m := make (map[string] map[string] int)
        subm := make (map[string] int)

        m["123"] = subm
        subm["abc"] = 1
        subm["bcd"] = 2

        if ch, ok := m["123"]; ok {
                fmt.Printf("%v\n", ch)  //ch is a map, not a chan
        }
}

[root@localhost test]# go build map.go   
[root@localhost test]# ./map 
map[bcd:2 abc:1]
```

Reply:
Oups, my bad, misread the type